### ISSUE TITLE
`TypeCheck<any>` results in `TS2589: Type instantiation is excessively deep and possibly infinite.`

### ISSUE THREAD

**Issue by SimenB** [sinclairzx81/typebox#798]
```ts
import type { TypeCheck } from '@sinclair/typebox/compiler';

const checker: TypeCheck<any> = {} as any;

checker.Decode({});
```

While reducing the reproduction I figured out I could use `TypeCheck<TUnknown>` which works, but I thought to still report it 🙂

**Comment by sinclairzx81** [comment]
Hi @SimenB! :)

Ah, this is a good catch actually. I just published a quick fix for this on 0.32.17. In the update, passing `<any>` for the generic parameter is permitted (as it does technically pass the `TSchema` constraint), however the StaticDecode will infer as `unknown` if it detects `any` is being passed. I've drafted a bit info on the fix over at https://github.com/sinclairzx81/typebox/pull/799

Let me know if this helps resolve things :)
Cheers!
S
